examples/trk_subsampler.rs

```diff
@@ -42,15 +44,37 @@ fn main() {
 
     if let Ok(percent) = args.get_str("--percent").parse::<f32>() {
         let percent = percent / 100.0;
-        let mut rng = rand::thread_rng();
+        let mut rng = rand::weak_rng();
 
         for streamline in reader.into_iter() {
             if rng.gen::<f32>() < percent {
                 writer.write(&streamline);
             }
         }
-    } else if let Ok(_nb) = args.get_str("--number").parse::<usize>() {
-        panic!("Not implemented yet");
+    } else if let Ok(nb) = args.get_str("--number").parse::<usize>() {
+        if let Ok(seed) = args.get_str("--seed").parse::<u8>() {
+            let size = reader.header.nb_streamlines;
+            let mut rng = rand::XorShiftRng::from_seed([seed; 16]);
+
+            let mut sampled_indices = rand::seq::sample_indices(&mut rng, size, nb);
+            sampled_indices.sort();
+
+            let mut iter_indices = sampled_indices.into_iter();
+            let mut index = iter_indices.next();
+
+            for (idx, streamline) in reader.into_iter().enumerate() {
+                if let Some(index_to_save) = index {
+                    if idx == index_to_save {
+                        writer.write(&streamline);
+                        index = iter_indices.next();
+                    }
+                } else {
+                    break;
+                }
+            }
+        } else {
+            panic!("--number need a seed to be deterministic.");
```

Comment:
I was about to write: look at `Error::expect()` to avoid this `else panic` but I would prefer adding an option instead. Remove the "Deterministic" promise in docopt and use the seed only if a --seed was given.

Reply:
Do you want also the `--percent` to be deterministic if a seed is given?

Reply:
Yes, good idea, please do that.